contracts/VToken.sol

```diff
@@ -330,15 +337,22 @@ contract VToken is Ownable2StepUpgradeable, AccessControlled, VTokenInterface, E
 
     /**
      * @notice Accrues interest and reduces reserves by transferring to the protocol reserve contract
+     * @dev Gracefully return if reserves already reduced in accrueInterest
      * @param reduceAmount Amount of reduction to reserves
+     * @param kind kind of reserve
      * @custom:event Emits ReservesReduced event; may emit AccrueInterest
      * @custom:error ReduceReservesCashNotAvailable is thrown when the vToken does not have sufficient cash
      * @custom:error ReduceReservesCashValidation is thrown when trying to withdraw more cash than the reserves have
      * @custom:access Not restricted
      */
-    function reduceReserves(uint256 reduceAmount) external override nonReentrant {
+    function reduceReserves(uint256 reduceAmount, IProtocolShareReserve.IncomeType kind)
+        external
+        override
+        nonReentrant
+    {
         accrueInterest();
-        _reduceReservesFresh(reduceAmount);
+        if (reduceReservesBlockNumber == _getBlockNumber()) return;
```

Comment:
Why do we have this check?

Reply:
If reserves are already reduced in `accrueInterest` we should fail it gracefully that's why we need this check